Document inclusion criteria for new functionality in core developer guide (#6488)
scikit-image/scikit-image: doc/source/core_developer.md

   tests not only execute the code, but explores corner cases.  It is tempting
   not to review tests, but please do so.

> 5. **Licensing:** New contributions _must_ be compatible with  `scikit-image`'s 

Reviewer: ```suggestion
5. **Licensing:** New contributions _must_ be compatible with `scikit-image`'s 
```

>    license. Which licenses are compatible is a hard question to answer. If in

Reviewer: Link to the `LICENSE.txt` file?

Author: Thanks, I'm on it but conda is taking its sweet time to install the dependencies to test it locally. :sweat_smile:

>    hard question to answer. If in doubt, ask the team for help. If the 

Reviewer 2: How about _(...) contributions should be available in the same license or be compatible with (...) question to answer. Examples of BSD-compatible licenses are MIT (link) and Apache (link). If in doubt (...)_

Reviewer: +1 to @alexdesiqueira's suggestion; maybe "in the same license" --> "under the same license"

   contributor is not the copyright holder of the submitted code, ask for
   approval and include them in our license.

6. **Established methods:** In general, we are looking to include well cited 

>    algorithms and methods that are of general use to the imaging community. 

Reviewer 2: How about _(...) to include widely used algorithms and methods (...)_

   tests not only execute the code, but explores corner cases.  It is tempting
   not to review tests, but please do so.

5. **Licensing:** New contributions should be available under the same license

>    or be compatible with {doc}`scikit-image's license <license>`. Which ones are
>    compatible is a hard question to answer. Examples of BSD-compatible licenses

Reviewer 2: ```suggestion
   or be compatible with {doc}`scikit-image's license <license>`. Examples of BSD-
   compatible licenses
```

>    are the [MIT License][mit_license] and [Apache License 2.0][apache_2-2]. If
>    in doubt, ask the team for help. If the contributor is not the copyright

Reviewer 2: ```suggestion
   are the [MIT License][mit_license] and [Apache License 2.0][apache_2-2]. When
   in doubt, ask the team for help. If you, as the contributor, is not the copyright
```

>    holder of the submitted code, ask for approval and include them in our
>    license.

Reviewer 2: ```suggestion
   holder of the submitted code, please ask for approval from the original authors and
   include their names in `LICENSE.txt`. You can use the other entries in that file as
   templates.
```

6. **Established methods:** In general, we are looking to include algorithms and
   methods that are established and widely used in literature and by the imaging 
   community. While this is not a hard requirement, new contributions should

>    further {doc}`our mission <values>`.

Reviewer 2: ```suggestion
   be consistent with {doc}`our mission <values>`.
```
